app/helpers/reports_helper.rb

```diff
@@ -41,6 +43,88 @@ def render_report_element(element, provided_locals = nil)
     locals = provided_locals.nil? ? {} : provided_locals.clone
     locals[:children] = children_html
 
+    if provided_locals[:export_all]
+      # Set path and filename local variables for files and tables
+
+      if element['type_of'] == 'my_module_repository'
+        obj_name = Repository.find(element[:repository_id]).name + '.csv'
+        obj_folder_name = 'Inventories'
+
+        locals[:filename] = obj_name
+        locals[:path] = "#{obj_folder_name}/#{obj_name}"
+      elsif element['type_of'].in? %w(step_asset step_table result_asset
+                                      result_table)
+
+        parent_el = ReportElement.find(element['parent_id'])
+        parent_type = parent_el[:type_of]
+        parent = parent_type.singularize.classify.constantize
+                            .find(parent_el["#{parent_type}_id"])
+
+        if parent.class == Step
+          obj_name = if element['type_of'] == 'step_asset'
+                       name = Asset.find(element[:asset_id]).file_file_name
+                       suffix = name.split('.').second
+                       suffix.prepend('.') if suffix
+                       name.split('.').first
+                     else
```

Comment:
I would use here `elsif element['type_of'] == 'step_table'` to be more explicit.